src/mono/msbuild/apple/data/ProxyProjectForAOTOnHelix.proj

```diff
@@ -89,6 +90,13 @@
       <RuntimeComponents Include="marshal-ilgen" />
     </ItemGroup>
 
+    <ItemGroup Condition="'$(EnableAggressiveTrimming)' == 'true' and '$(UseNativeAOTRuntime)' != 'true'">
+       <TrimmerRootDescriptor Include="$(ExtraFilesPath)**\*.xml" />
+
+       <!-- To recreate the state of feature flags (RuntimeHostConfigurationOption) during app build-time we need to discard the default setting, coming from SDK, and used the values passed in ProxyProjectForAOTOnHelix.props -->
+       <RuntimeHostConfigurationOption Remove="@(_AppleUsedRuntimeHostConfigurationOption)" />
+       <RuntimeHostConfigurationOption Include="@(_AppleUsedRuntimeHostConfigurationOption)" />
```

Comment:
Should this be
```suggestion
       <RuntimeHostConfigurationOption Remove="@(RuntimeHostConfigurationOption)" />
       <RuntimeHostConfigurationOption Include="@(_AppleUsedRuntimeHostConfigurationOption)" />
```

Reply:
I will double check, but the idea is to only overwrite the configuration passed on from build machines, while keeping what has been configured for build on helix.

Reply:
Maybe I'm missing something, but the original change to me looks like it functionally does nothing? A - B + B?

Reply:
Ahh that makes more sense to me. Thank you for adding the example! I totally missed that it would be helpful for updating the metadata values

Reply:
I double-checked this and it is required since some of the feature switches can be specified on the test project level and they will be sent to Helix in form of `_AppleUsedRuntimeHostConfigurationOption`. On the other hand, we cannot *just* use `_AppleUsedRuntimeHostConfigurationOption` instead of `RuntimeHostConfigurationOption` as it does not include the default configuration coming from the SDK.